plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -162,10 +186,44 @@ private void processPredictions(
         List<Map<String, String>> processOutputMap,
         int inputMapIndex,
         int inputMapSize
-    ) {
+    ) throws IOException {
         Map<String, String> modelParameters = new HashMap<>();
+        Map<String, String> modelConfigs = new HashMap<>();
+
         if (inferenceProcessorAttributes.getModelConfigMaps() != null) {
             modelParameters.putAll(inferenceProcessorAttributes.getModelConfigMaps());
+            modelConfigs.putAll(inferenceProcessorAttributes.getModelConfigMaps());
+        }
+        Map<String, String> outputMapping = processOutputMap.get(inputMapIndex);
+
+        Map<String, Object> ingestDocumentSourceAndMetaData = new HashMap<>();
+        ingestDocumentSourceAndMetaData.putAll(ingestDocument.getSourceAndMetadata());
+        ingestDocumentSourceAndMetaData.put(IngestDocument.INGEST_KEY, ingestDocument.getIngestMetadata());
+
+        Map<String, List<String>> newOutputMapping = new HashMap<>();
+        for (Map.Entry<String, String> entry : outputMapping.entrySet()) {
+            String newDocumentFieldName = entry.getKey();
+            List<String> dotPathsInArray = writeNewDotPathForNestedObject(ingestDocumentSourceAndMetaData, newDocumentFieldName);
+            newOutputMapping.put(newDocumentFieldName, dotPathsInArray);
+        }
+
+        for (Map.Entry<String, String> entry : outputMapping.entrySet()) {
+            String newDocumentFieldName = entry.getKey();
+            List<String> dotPaths = newOutputMapping.get(newDocumentFieldName);
+
+            int existingFields = 0;
+            for (String path : dotPaths) {
+                if (ingestDocument.hasField(path)) {
+                    existingFields++;
```

Comment:
when the document doesn't have the new field, will it add the newField to output mapping?